lib/internal/Magento/Framework/App/DeploymentConfig.php

```diff
@@ -247,37 +250,43 @@ private function getAllEnvOverrides(): array
      */
     private function flattenParams(array $params, ?string $path = null, array &$flattenResult = null): array
     {
-        if (null === $flattenResult) {
-            $flattenResult = [];
-        }
+        $paramsKey = CRC32(json_encode($params ?? ''));
```

Comment:
```suggestion
        $paramsKey =  empty($params) ? '2d06800538d394c2' : hash('xxh3', json_encode($params));
```
it shold be even better from perfromance point of view. see https://php.watch/versions/8.1/xxHash

Reply:
![image](https://github.com/magento/magento2/assets/4256191/2e1f2eb1-7250-47ee-aaa0-4cd1472618e1)
thanks to this, we get 10% improvement 

Great tip, thanks!

Reply:
Because $params is expected to be an array, we could avoid to use the empty function and compare to [] or use !$params instead (it's fine tunning, it won't bring incredible improvement)

Reply:
if #params is empty, you could return the result directly also, without hashing.

```
if (!$params) {
   return $flattenResult ?? [];
}
```